src/main/java/org/jfrog/bamboo/task/ArtifactoryDotNetTaskBase.java

```diff
@@ -0,0 +1,187 @@
+import org.jfrog.bamboo.admin.ServerConfig;
+import org.jfrog.bamboo.builder.BuildInfoHelper;
+import org.jfrog.bamboo.configuration.BuildParamsOverrideManager;
+import org.jfrog.bamboo.context.DotNetBuildContext;
+import org.jfrog.bamboo.util.BuildInfoLog;
+import org.jfrog.bamboo.util.TaskUtils;
+import org.jfrog.build.api.Artifact;
+import org.jfrog.build.api.Build;
+import org.jfrog.build.extractor.clientConfiguration.ArtifactoryBuildInfoClientBuilder;
+import org.jfrog.build.extractor.clientConfiguration.ArtifactoryDependenciesClientBuilder;
+import org.jfrog.build.extractor.clientConfiguration.util.spec.SpecsHelper;
+import org.jfrog.build.extractor.nuget.extractor.NugetRun;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by Bar Belity on 14/10/2020.
+ */
+public abstract class ArtifactoryDotNetTaskBase extends ArtifactoryTaskType {
+
+    protected final EnvironmentVariableAccessor environmentVariableAccessor;
+    protected final CapabilityContext capabilityContext;
+    protected CustomVariableContext customVariableContext;
+    protected DotNetBuildContext dotNetBuildContext;
+    protected BuildInfoHelper buildInfoHelper;
+    protected BuildParamsOverrideManager buildParamsOverrideManager;
+    protected Map<String, String> environmentVariables;
+    protected Path workingDir;
+    protected String buildName;
+    protected String buildNumber;
+    protected TaskType taskType;
+    protected String specTemplate = "{\n" +
+            "  \"files\": [\n" +
+            "    {\n" +
+            "      \"pattern\": \"%s\",\n" +
+            "      \"target\": \"%s\"\n" +
+            "    }\n" +
+            "  ]\n" +
+            "}";
+
+    protected void initTask(CommonTaskContext context, String taskKey, String executable, String taskName) throws TaskException {
+        super.initTask(context);
+        dotNetBuildContext = new DotNetBuildContext(taskContext.getConfigurationMap());
+        buildParamsOverrideManager = new BuildParamsOverrideManager(customVariableContext);
+        BuildContext buildContext = ((TaskContext) context).getBuildContext();
+        buildName = dotNetBuildContext.getBuildName(buildContext);
+        buildNumber = dotNetBuildContext.getBuildNumber(buildContext);
+        initBuildInfoHelper(buildContext);
+        environmentVariables = getEnv(taskKey, executable, taskName);
+        workingDir = getWorkPath();
+    }
+
+    @NotNull
+    @Override
+    public TaskResult runTask(@NotNull TaskContext taskContext) {
+        Build build;
+        if (dotNetBuildContext.isRestoreCommand()) {
+            build = executeRestore();
+        } else {
+            try {
+                build = executePush();
+            } catch (Exception e) {
+                String message = "Exception occurred while executing task";
+                logger.addErrorLogEntry(message, e);
+                log.error(message, e);
```

Comment:
`buildInfoLog.error(...)` does exactly the same
```suggestion
                buildInfoLog.error("Exception occurred while executing task", e);
```

Reply:
Fixed.